x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_logic.ts

```diff
@@ -100,16 +125,49 @@ export const EngineLogic = kea<MakeLogicType<EngineValues, EngineActions>>({
     ],
   }),
   listeners: ({ actions, values }) => ({
-    initializeEngine: async () => {
+    initializeEngine: async (_, breakpoint) => {
+      breakpoint(); // Prevents errors if logic unmounts while fetching
+
       const { engineName } = values;
       const { http } = HttpLogic.values;
 
       try {
         const response = await http.get(`/api/app_search/engines/${engineName}`);
         actions.setEngineData(response);
       } catch (error) {
-        actions.setEngineNotFound(true);
+        if (error?.response?.status >= 400 && error?.response?.status < 500) {
+          actions.setEngineNotFound(true);
+        } else {
+          flashErrorToast(POLLING_ERROR_TITLE, {
+            text: POLLING_ERROR_TEXT,
+            toastLifeTimeMs: POLLING_DURATION * 0.75,
+          });
```

Comment:
Here's a screencap of this toast behavior:

![error_toast](https://user-images.githubusercontent.com/549407/123029567-30126880-d396-11eb-9aca-a1a2fcdc8409.gif)

Quick UX note: This toast won't appear when Enterprise Search is down (because then the app 502s and the generic Error Connecting view gets shown). Instead, this toast occurs when **Kibana** is somehow down and the user has a stale/cached frontend (hence the screencap briefly showing my terminal, where I stop Kibana). It's fairly unlikely and 404/401 redirects are going to be the most common case, but I thought I should add in something as a generic 5xx catch.